orc8r/tools/magma-manipulator/magma_manipulator/magma_api.py

```diff
@@ -0,0 +1,486 @@
+"""
+Copyright (c) 2016-present, Facebook, Inc.
+All rights reserved.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree. An additional grant
+of patent rights can be found in the PATENTS file in the same directory.
+"""
+
+import json
+import logging
+import requests
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+from urllib.parse import urljoin
+
+from magma_manipulator import exceptions
+
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+LOG = logging.getLogger(__name__)
+
+
+def is_network_exist(orc8r_api_url, gw_net, certs):
+    LOG.info('Check if network {gw_net} exists'.format(gw_net=gw_net))
+    magma_net_url = urljoin(orc8r_api_url,
+                            'magma/v1/networks/{gw_net}'.format(gw_net=gw_net))
+    LOG.debug('Make get request to {url}'.format(url=magma_net_url))
+    resp = requests.get(magma_net_url, verify=False, cert=certs)
+    str_result = resp.content.decode('ascii')
+    json_result = json.loads(str_result)
+    LOG.debug('Received result {result}'.format(result=json_result))
+    if 'id' in json_result:
+        return json_result['id'] == gw_net
+    return False
+
+
+def create_network(orc8r_api_url, gw_net, certs):
+    LOG.info('Start to create network {gw_net}'.format(gw_net=gw_net))
+    magma_net_url = urljoin(
+        orc8r_api_url,
+        'magma/v1/networks')
+    data = {
+        'description': 'This network created from automation tool',
+        'dns': {
+          'enable_caching': False,
+          'local_ttl': 0,
+        },
+        'id': gw_net,
+        'name': gw_net
+      }
+
+    headers = {'content-type': 'application/json',
+               'accept': 'application/json'}
+    resp = requests.post(magma_net_url,
+                         data=json.dumps(data),
+                         headers=headers,
+                         verify=False,
+                         cert=certs)
+    msg = 'Receive response {text} with status code '\
+          '{status_code} afte network {gw_net} creation.'.format(
+                  text=resp.text,
+                  status_code=resp.status_code,
+                  gw_net=gw_net)
+    LOG.info(msg)
+    if resp.status_code not in [200, 201, 204]:
+        raise exceptions.MagmaRequestException(msg)
+
+
+def _get_register_gateway_url(gw_net_type, gw_net):
+    if gw_net_type == 'carrier_wifi_network':
+        return 'magma/v1/cwf/{gw_net}/gateways'.format(gw_net=gw_net)
+    elif gw_net_type == 'feg':
+        return 'magma/v1/feg/{gw_net}/gateways'.format(gw_net=gw_net)
+
+
+def _get_register_gateway_data(gw_net_type, gw_id, gw_uuid, gw_key, gw_name):
```

Comment:
We should be able to get rid of this entire function by just registering the gateway with an existing config. See my comment below.